app/src/main/java/io/pslab/activity/CreateConfigActivity.java

```diff
@@ -0,0 +1,170 @@
+package io.pslab.activity;
+
+import android.os.Bundle;
+import android.os.Environment;
+import android.support.design.widget.Snackbar;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.util.SparseBooleanArray;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ListView;
+import android.widget.Spinner;
+import android.widget.Toast;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+
+import io.pslab.R;
+import io.pslab.others.CSVLogger;
+import io.pslab.others.CustomSnackBar;
+
+public class CreateConfigActivity extends AppCompatActivity {
+
+    private ArrayList<String> instrumentsList;
+    private ArrayList<String[]> instrumentParamsList;
+    private ListView paramsListView;
+    private int selectedItem = 0;
+    private String intervalUnit = "sec";
+    private EditText intervalEditText;
+    private String interval;
+    private View rootView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_create_config);
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        rootView = findViewById(R.id.create_config_root_view);
+        paramsListView = findViewById(R.id.params_list);
+        Spinner selectInstrumentSpinner = findViewById(R.id.select_instrument_spinner);
+        Spinner intervalUnitSpinner = findViewById(R.id.interval_unit_spinner);
+        intervalEditText = findViewById(R.id.interval_edit_text);
+        Button createConfigFileBtn = findViewById(R.id.create_config_btn);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setTitle(R.string.nav_config);
+        if (getSupportActionBar() != null) {
+            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+            getSupportActionBar().setDisplayShowHomeEnabled(true);
+        }
+        instrumentsList = new ArrayList<>();
+        instrumentParamsList = new ArrayList<>();
+        createArrayLists();
+        selectInstrumentSpinner.setAdapter(new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, instrumentsList));
+        selectInstrumentSpinner.setSelection(0, true);
+        createCheckboxList();
+        selectInstrumentSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                selectedItem = position;
+                createCheckboxList();
```

Comment:
Why not setting position as a parameter to `createCheckboxList()` method and use it directly inside the method?

`createCheckboxList(position);`

Reply:
I am again using ``` selectedItem ``` variable in ``` onClick() ``` method of ``` createConfigBtn``` and also in the ``` createConfigFile() ``` function.